lib/GitterUser.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+var Promise = require("bluebird");
+
+var rp = require("request-promise");
+
+// miliseconds of interval between presence=online updates
+var ONLINE_PERIOD = 4 * 60 * 1000;
+
+// miliseconds of grace period before users count as offline
+var OFFLINE_GRACE = 3 * 60 * 1000;
+
+function GitterUser(bridge, remoteUserModel) {
+  this._bridge = bridge;
+  this._remoteUserModel = remoteUserModel;
+
+  this._presenceRooms = {};
```

Comment:
I assume this is going to be a map. I hope it's a map. Maps are fun. They have keys! Shame I don't know what the keys are without grepping the entire class....

(comment please!)